### PR TITLE
[1.2] Remove legacy asset key format from queries

### PR DESCRIPTION
**Pull request by clairelin135**
Fixes https://github.com/dagster-io/dagster/issues/10240

This PR removes support for the legacy asset key serialization (path joined by `'.'`). Users who were using assets prior to `0.10.0` may be affected. For these users, we'll direct them to reach out to us for a data migration.

### CODE REVIEW

**Review by sryza — APPROVED** [review]
Nice - let's wait until the branch cut to merge this and make sure to include a note in the migration guide.

**Review by prha — APPROVED** [review]

**Pull request merged**
